sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -1309,6 +1318,15 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     client.listFunctions(db, pattern)
   }
 
+  def isResourceChanged(db: String, newFunction: CatalogFunction): Boolean = {
+    val oldFunction = getFunction(db, newFunction.identifier.funcName)
+    if (oldFunction.resources.size != newFunction.resources.size) {
+      return true
+    }
+    val changedResources = newFunction.resources.filter(
+      newResource => !oldFunction.resources.contains(newResource))
```

Comment:
Will case sensitivity be problem in this comparison? Shall we lower case before comparison?

Reply:
I think it should be case sensitive, cz in case of unix based system  /opt/some.jar is different from /OPT/SOME.jar. Isn't it ?